lib/truffle/truffle/patching.rb

```diff
@@ -78,4 +78,21 @@ def activate
     end
   end
 
+  # Allows specifying patching outside the context of RubyGems by setting an environment variable value.
+  # E.g., setting TRUFFLERUBY_CUSTOM_PATCH='launchy:lib,spec' will allow patching of files in the 'lib' and 'spec'
+  # of a local checkout of the 'launchy' gem.
+  def install_local_patches
+    custom_patch = Truffle::POSIX::NATIVE && ENV['TRUFFLERUBY_CUSTOM_PATCH'] # We can't read from ENV if native is disabled.
+
+    if custom_patch
+      name, paths = custom_patch.split(':')
+      begin
+        Truffle::Patching.insert_patching_dir name, *paths.split(',')
+      rescue # rubocop:disable Lint/HandleExceptions
```

Comment:
What kind of exception happens here? Can you show a backtrace?
Shouldn't it work in all situations if TRUFFLERUBY_CUSTOM_PATCH is not erroneous and the Ruby home is known?

Reply:
The comment below details the exact failure scenario. I run `rake test` and that parent process has nothing to patch (but the env var is set). `rake test` then spawns a new Ruby process that runs the tests with the library that should be patched. I don't want to have to deal with trying to patch Rakefiles, so it's considerably more convenient to just set the env var at the outer level and allow patching to fail.

The exception is the one from `insert_patching_dir`: `raise "Could not find paths #{paths} in $LOAD_PATH (#{$LOAD_PATH})" unless insertion_point`.

Reply:
I see, I was thinking we could add the paths in $LOAD_PATH here so we're sure to find them, like:
```ruby
$LOAD_PATH.unshift(*paths)
```
But this might have unintended side effects.

Patching will not work if the process adds a path in front of $LOAD_PATH dynamically (`$LOAD_PATH.unshift(File.expand_path("../lib"))`) and not via `-I`, but the current approach doesn't support that either.

Reply:
Right. I'm making things work for a very narrow use case.

Reply:
I think at least printing a warning with the exception's message might be good, to avoid nasty surprises that something else broke silently.